app/js/app.js

```diff
@@ -68,9 +68,54 @@ if (console) {
               "font-size:1.5em;color:#4558c9;", "color:#d61a7f;font-size:1em;");
 }
 
+// Show a toast with a service-worker-related update.
+window.showToast = function(message) {
+  var toast = document.getElementById('sw-toast');
+
+  if (!toast) {
+    toast = document.createElement('paper-toast');
+    toast.id = 'sw-toast';
+    document.body.appendChild(toast);
+  }
+
+  toast.textContent = message;
+  toast.show();
+};
+
 // Register service worker if supported.
 if ('serviceWorker' in navigator) {
-  navigator.serviceWorker.register('/service-worker.js');
+  navigator.serviceWorker.register('/service-worker.js'
+  ).then(function(registration) {
+    registration.onupdatefound = function() {
+      // The updatefound event implies that registration.installing is set; see
+      // https://slightlyoff.github.io/ServiceWorker/spec/service_worker/index.html#service-worker-container-updatefound-event
+      const installingWorker = registration.installing;
+      installingWorker.onstatechange = function() {
+        switch (installingWorker.state) {
+          case 'installed':
+            if (!navigator.serviceWorker.controller) {
+              window.showToast('Caching complete! Future visits will work offline.');
```

Comment:
I'm not sure whether we should do this? Since we have runtime caching, only pages the user has visited will be available offline, so this seems like a bit of an oversell. 

(If we have a toast, maybe "Pages you view are cached for offline use."? Not wedded to this wording, just trying to find a middle ground.)